src/protocol/sort/multi_bit_permutation.rs

```diff
@@ -61,18 +61,22 @@ pub async fn multi_bit_permutation<'a, F: Field, S: SecretSharing<F>, C: Context
     }
 
     // Take sum of products of output of equality check and accumulated sum
-    let mut one_off_permutation =
-        try_join_all((0..num_records).zip(repeat(ctx)).map(|(rec, ctx)| {
-            let mut sop_inputs = Vec::with_capacity(num_possible_bit_values);
-            for idx in 0..num_possible_bit_values {
-                sop_inputs.push((&equality_checks[rec][idx], &prefix_sum[rec][idx]));
-            }
```

Comment:
All of this was eliminated. No need to create a new vector for every row, and push a bunch of pairs into it.